packages/itwinui-css/src/tree/tree.scss

```diff
@@ -2,19 +2,54 @@
 // See LICENSE.md in the project root for license terms and full copyright notice.
 @use '../mixins';
 
-$iui-active-outline: thin solid var(--iui-color-border-accent);
-
-// padding of small borderless button + 1px transparent border
-$iui-expander-inline-padding: calc(var(--iui-size-2xs) + 1px);
-
-// width of small borderless button (button is 1:1 ratio)
-$iui-expander-button-width: calc(var(--iui-component-height-small) + 2px);
+$iui-active-outline: 1px solid var(--iui-color-border-accent);
 
 .iui-tree {
   @include mixins.iui-reset;
+  --_iui-expander-button-width: calc(var(--iui-component-height-small) + 2px);
+  --_iui-tree-node-padding-inline: var(--iui-size-xs);
+  --_iui-tree-row-height: calc(var(--iui-size-3xs) * 18);
+
   list-style: none;
   border-radius: var(--iui-border-radius-1);
   background-color: var(--iui-color-background);
+
+  &[data-iui-size='small'] {
+    --_iui-tree-node-padding-inline: var(--iui-size-2xs);
+    --_iui-tree-row-height: calc(var(--iui-size-3xs) * 14);
+  }
+
+  &[data-iui-lines='true'] {
+    isolation: isolate;
+
+    &:hover,
+    &:focus-within {
+      --_iui-tree-line-color: var(--iui-color-border-subtle);
+    }
+
+    .iui-sub-tree {
+      position: relative;
+
+      &::before {
+        content: '';
+        inline-size: 1px;
+        block-size: 100%;
+        position: absolute;
+        inset: 0 auto 0
```

Comment:
can you explain what math is going on here? a little hard to follow.

Reply:
Oh gosh, let's see if I can remember. 😅

GitHub isn't showing the code, so here it is for reference:
```sass
::before {
  inset: 0 auto 0 calc(var(--_iui-expander-button-width) * 0.5 * var(--level, 0) + (var(--iui-size-m) + var(--_iui-tree-node-padding-inline) * 2));
}
```

`var(--_iui-tree-node-padding-inline) * 2` is to account for the padding on the tree body as well as the gap between checkbox & expander button.
![Screenshot 2023-12-05 at 5 29 57 PM](https://github.com/iTwin/iTwinUI/assets/849817/ac8a87db-e0cd-4798-954d-13c478864637) ![Screenshot 2023-12-05 at 5 30 18 PM](https://github.com/iTwin/iTwinUI/assets/849817/a67a2342-268f-46be-b29c-a6102a056364)

`var(--_iui-expander-button-width) * 0.5` is to account for the expander button, but only half of it, so the line is centered on the button.

`var(--level, 0) + (var(--iui-size-m)` is the indentation level as well as the checkbox which uses M width.

Will need to think up some way to accommodate for tree without checkboxes, because that won't work at the moment.

Reply:
To make the calculation easier to read, would it be better to create SASS variables only in this scope? You could then add comments for each variable to explain what it does.

So something along the lines of:
```sass
&::before {
  $_tree-body-padding-and-gap: calc(var(--_iui-tree-node-padding-inline) * 2) // explanation ...
  $_expander-button-width: var(--_iui-expander-button-width) * 0.5 // explanation ...
  $_indentation: var(--level, 0) + (var(--iui-size-m)) // explanation ...
  // ... other helper variables

  $_left-padding: combination of the helper variables
}
```

Reply:
Will do this in a separate PR addressing lines.  Resolving this comment to declutter this PR for now.